### PR TITLE
Snap 851

### PR DESCRIPTION
**Pull request by nthanvi**
## Changes proposed in this pull request
use default schema name "APP" when no schema is specified.
when user do set schema with thin client connection or during query routing scenarios then schema should be  maintain for the session.

it  also fixes SNAP-835  and   SNAP-233
## Patch testing
precheckin. 
changed few tests to have coverage for fullyQualifiedTable name
Added new test to check set schema with query routing scenarios

## Other PRs 
Yes
https://github.com/SnappyDataInc/snappy-store/pull/79
https://github.com/SnappyDataInc/snappy-aqp/pull/62

### CODE REVIEW

**Comment by nthanvi** [comment]
@ymahajan

File: core/src/test/scala/org/apache/spark/sql/store/ColumnTableTest.scala [hunk 1]
```diff
@@ -503,6 +497,62 @@ class ColumnTableTest
...
+    val rs = conn.createStatement().executeQuery("select count (*) from " + tableName)
+    if (rs.next()) {
+      //The row buffer should not have more than 2 rows as the batch size is 3
+      assert(rs.getInt(1) <= 2)
+    }
+
+    rs.close()
+    conn.close()
+
+  }
```
Comment (ymahajan) [inline core/src/test/scala/org/apache/spark/sql/store/ColumnTableTest.scala]: A test which mixes cases of with and without schema would be good. (with default APP schema)

File: core/src/test/scala/org/apache/spark/sql/store/ColumnTableTest.scala [hunk 2]
```diff
@@ -146,18 +146,12 @@ class ColumnTableTest
...
-    dataDF = snc.createDataFrame(rdd)
     dataDF.write.format("column").mode(SaveMode.Append).options(props).saveAsTable(tableName)
     result = snc.sql("SELECT * FROM " + tableName)
     r = result.collect
     assert(r.length == 11)
 
     // Overwrite if table is present
-    data = Seq(Seq(100, 200, 300), Seq(700, 800, 900), Seq(900, 200, 300), Seq(400, 200, 300), Seq(500, 600, 700), Seq(800, 900, 1000))
-    rdd = sc.parallelize(data, data.length).map(s => new Data(s(0), s(1), s(2)))
-    dataDF = snc.createDataFrame(rdd)
```
Comment (ymahajan) [inline core/src/test/scala/org/apache/spark/sql/store/ColumnTableTest.scala]: looks like repeated dataframe with same data is removed.

File: core/src/main/scala/org/apache/spark/sql/store/StoreUtils.scala [hunk 3]
```diff
@@ -316,4 +317,4 @@ object StoreUtils extends Logging {
       true
     })
   }
-}
+}
```
Comment (rishitesh) [inline core/src/main/scala/org/apache/spark/sql/store/StoreUtils.scala]: Unnecessary change.

File: cluster/src/dunit/scala/io/snappydata/cluster/SplitSnappyClusterDUnitTest.scala [hunk 4]
```diff
@@ -140,7 +140,7 @@ object SplitSnappyClusterDUnitTest extends SplitClusterDUnitTestObject {
     val snc = SnappyContext(sc)
     // remove below once SNAP-653 is fixed
     val numPartitions = props.getOrElse("buckets", "113").toInt
-    StoreUtils.removeCachedObjects(snc, "EMBEDDEDMODETABLE1", numPartitions,
+    StoreUtils.removeCachedObjects(snc, "APP.EMBEDDEDMODETABLE1", numPartitions,
```
Comment (rishitesh) [inline cluster/src/dunit/scala/io/snappydata/cluster/SplitSnappyClusterDUnitTest.scala]: Why we needed the APP here ?

File: core/src/dunit/scala/io/snappydata/cluster/SplitClusterDUnitTestBase.scala [hunk 5]
```diff
@@ -191,8 +188,9 @@ trait SplitClusterDUnitTestObject {
 
     // remove below once SNAP-653 is fixed
     val numPartitions = props.getOrElse("buckets", "113").toInt
-    StoreUtils.removeCachedObjects(snc, "SPLITMODETABLE1", numPartitions,
+    StoreUtils.removeCachedObjects(snc, "APP.SPLITMODETABLE1", numPartitions,
```
Comment (rishitesh) [inline core/src/dunit/scala/io/snappydata/cluster/SplitClusterDUnitTestBase.scala]: See my comments above, why we need the APP.

File: core/src/test/scala/org/apache/spark/sql/store/ColumnTableTest.scala (hunk 1 above)
Comment (rishitesh) [inline core/src/test/scala/org/apache/spark/sql/store/ColumnTableTest.scala]: +1 to Yogesh

Comment (rishitesh) [inline core/src/test/scala/org/apache/spark/sql/store/ColumnTableTest.scala]: Rest all looks fine.

Comment (nthanvi) [inline core/src/test/scala/org/apache/spark/sql/store/ColumnTableTest.scala]: Already added coverage for a  test with and without schema

File: cluster/src/dunit/scala/io/snappydata/cluster/SplitSnappyClusterDUnitTest.scala (hunk 4 above)
Comment (nthanvi) [inline cluster/src/dunit/scala/io/snappydata/cluster/SplitSnappyClusterDUnitTest.scala]: it is a product  utility method  which takes the qualifiedTableName and now the table name is always going to be fullyQualified even if user provides a table name without schema.
